lang/semgrep-grammars/src/semgrep-dockerfile/test/corpus/semgrep.txt

```diff
@@ -65,3 +65,380 @@ RUN ["a", $X, "b"]
       (double_quoted_string)
       (semgrep_metavariable)
       (double_quoted_string))))
+
+=============================================
+Param metavariable
+=============================================
+
+FROM --platform=$PLATFORM debian
+ADD --chown=$USER_GROUP src dst
+
+---
+
+(source_file
+  (from_instruction
+    (param)
+    (image_spec
+      (image_name)))
+  (add_instruction
+    (param)
+    (path)
+    (path)))
+
+=============================================
+Image metavariable
+=============================================
+
+FROM $NAME:$TAG@$DIGEST
+FROM ${NAME}:${TAG}@${DIGEST}
+
+---
+
+(source_file
+  (from_instruction
+    (image_spec
+      (image_name
+        (expansion
+          (variable)))
+      (image_tag
+        (expansion
+          (variable)))
+      (image_digest
+        (expansion
+          (variable)))))
+  (from_instruction
+    (image_spec
+      (image_name
+        (expansion
+          (variable)))
+      (image_tag
+        (expansion
+          (variable)))
+      (image_digest
+        (expansion
+          (variable))))))
+
+=============================================
+Image alias metavariable
+=============================================
+
+FROM debian AS $ALIAS
+
+---
+
+(source_file
+  (from_instruction
+    (image_spec
+      (image_name))
+    (image_alias
+      (expansion
+        (variable)))))
+
+=============================================
+File metavariables
+=============================================
+
+ADD $A $B $DST
+COPY $A $B $DST
+VOLUME $A
+
+---
+
+(source_file
+  (add_instruction
+    (path
+      (expansion
+        (variable)))
+    (path
+      (expansion
+        (variable))
+      (expansion
+        (variable))))
+  (copy_instruction
+    (path
+      (expansion
+        (variable)))
+    (path
+      (expansion
+        (variable))
+      (expansion
+        (variable))))
+  (volume_instruction
+    (path
+      (expansion
+        (variable)))))
+
+=============================================
+File list ellipsis
+=============================================
+
+ADD ... dst
+COPY ... dst
+VOLUME ...
+
+---
+
+(source_file
+  (add_instruction
+    (path)
```

Comment:
Ellipsis is reduced to 'path'? Is it intended too? Shouldn't we have a special (ellipsis) construct in the CST?

Reply:
`...` is a valid path, so again we need to inspect the token when translating a pattern and convert it to an ellipsis if its value is `...`. When parsing a dockerfile target, `...` will remain a string.

Reply:
This could be a useful comment to put in the test file.